sql/password.c

```diff
@@ -122,7 +122,14 @@ void my_make_scrambled_password_323(char *to, const char *password,
 {
   ulong hash_res[2];
   hash_password(hash_res, password, (uint) pass_len);
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
   sprintf(to, "%08lx%08lx", hash_res[0], hash_res[1]);
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
 }
```

Comment:
At least one caller (`execute_commands()` in `client/mysqladmin.cc`) is invoking this with `to` pointing to a buffer of 64 bytes. Based on the format expression, the length should be 16 bytes. Based on the `ulong` data type, whose size is 64 bits on LP64 systems and 32 bits elsewhere (including the LLP64 Microsoft Windows), the length could be up to 32 bytes. It seems to me that the `ulong` should be replaced with `uint32_t`, because `SCRAMBLE_LENGTH_323` is defined as 8.

In `Item_func_password::val_str_ascii()`, the argument is a pointer to the buffer `tmp_value[SCRAMBLED_PASSWORD_CHAR_LENGTH+1]`. It seems that this is the length that should be passed to `snprintf` here, and should be mentioned in the function documentation as well.